src/test/java/org/opensearch/neuralsearch/processor/RRFProcessorIT.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor;
+
+import lombok.SneakyThrows;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.index.query.MatchQueryBuilder;
+import org.opensearch.neuralsearch.BaseNeuralSearchIT;
+import org.opensearch.neuralsearch.query.HybridQueryBuilder;
+import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.opensearch.neuralsearch.util.TestUtils.DELTA_FOR_SCORE_ASSERTION;
+
+public class RRFProcessorIT extends BaseNeuralSearchIT {
+
+    private int currentDoc = 1;
+    private static final String RRF_INDEX_NAME = "rrf-index";
+
+    @SneakyThrows
+    public void testRRF_whenValidInput_thenSucceed() {
+        String ingestPipelineName = "rrf-ingest-pipeline";
+        String modelId = null;
+        try {
+            modelId = prepareModel();
```

Comment:
you don't need to deploy model, I think you can get away with knn query only. We should try to avoid unnecessary model uploads because this increases chance on flaky tests